### DISCUSSION TITLE
[Feature] Keyboard shortcuts

### DISCUSSION THREAD

**Discussion by JustAmply** [immich-app/immich#3085] [Feature Request]
### The feature

Immich would greatly benefit from the inclusion of keyboard shortcuts. As users often perform repetitive tasks, having the ability to execute these actions through intuitive keyboard shortcuts would significantly enhance their workflow.

To my knowledge there already exist shortcuts for switching to the next or previous image (left and right arrow key) and to exit the full screen view (ESC).

But shortcuts for marking as a favourite, archiving or downloading are still missing. 
These would greatly improve the productivity and experience of (power) users.

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by prometheanfire** [comment]
when browsing, h,j,k,l for moving and x to select, maybe d to download would be nice.  similar to vim keybindings

**Comment by Rdkang** [comment]
Just wanted to let you know that when you're viewing an image, you can use the 'f' key to toggle the favorite status

**Reply by Flopet** [reply]
This is exactly what I was looking for, thank you!

**Comment by prometheanfire** [comment]
is there a commit this was fixed in? (I see that the issue was closed)

**Comment by mstewart14** [comment]
I'd appreciate having the ability to navigate an album page or the photos page using the keyboard as well (page up, page down, arrow keys up and down).

**Comment by bo0tzz** [comment]
Keyboard shortcuts were added a while ago, you can press `?`  on the web UI to bring up an overview.

**Reply by mstewart14** [reply]
Much appreciated, I have added an [idea ](https://github.com/immich-app/immich/discussions/6202) for this specifically.  I do understand there are keyboard shortcuts (mostly by reading these notes, I did eventually find the [documentation_](https://immich.app/blog/2023/07/29/update#shortcuts-for-common-actions-on-the-web)).  I've also proposed another [idea ](https://github.com/immich-app/immich/discussions/6203) to enable configuration of those shortcuts.  Thank you for all the work you guys have done on Immich... it is awe inspiring and I am deeply grateful.

**Comment by prometheanfire** [comment]
nice, at least it works in single photo viewing

I assume this would be a new feature discussion, but from the gallery view in web, navigating by keyboard shortcuts, being able to select multiple photos (not just a range, but a random selection), then doing a bulk action (delete, archive, etc) is what I was looking for.

Some of that seems like shortcuts (x to select from gallery view, navigation via keyboard in gallery view).  The last part may be a new request though (bulk actions).

**Comment by psla** [comment]
Immich is amazing.

One shortcut I wish was present is "add this picture/video to the latest album" (latest as defined by latest created, or as defined as "last selected"). (Similarly, adding pictures to albums in the android app is sub-optimal: usually my newly created album shows up last, so I need to click "add to album", then scroll all the way down to the newest album).

**Reply by nghiemlong** [reply]
I would like to have shortcut for "Refreshing faces" while viewing an image. I see many pictures in which faces did not recognized.

**Comment by RobinEd** [comment]
you can pgup / pgdn and arrow up/down through the album / folders / photos thumbnail views, but you need to select a thumbnail first. It would be nice if this was also indicated in the ?

**Comment by shadowlocked** [comment]
It would be great if shortcuts (such as 'Fix Incorrect Match') could be assigned, as well as others in a tagging or curation context.
